docs/design/features/DynamicPgo.md

```diff
@@ -0,0 +1,693 @@
+# Dynamic PGO
+
+The following document outlines the overall approach to profile guided optimization (aka PGO) that I'd like to see us adopt in the jit.
+
+It is based on past experience and familiarity with profile guided optimization in several different AOT compilation toolchains, namely
+* HP-UX Compilers (circa 1993-1997, known as PBO)
+* MSVC's implementation of PGO (and PGO2)
+* Midori's implementation of PGO in Bartok/Phoenix
+* LLVM's implementation of PGO (known as FDO)
+
+However, the approach described here differs materially from those efforts because the primary focus of this work is a jitted environment. As such, it presents unique challenges and compelling new opportunities. That being said,much of the work proposed below will also benefit AOT scenarios.
```

Comment:
```suggestion
However, the approach described here differs materially from those efforts because the primary focus of this work is a jitted environment. As such, it presents unique challenges and compelling new opportunities. That being said, much of the work proposed below will also benefit AOT scenarios.
```